### PR TITLE
Improved merge-base-class transformation

### PR DESCRIPTION
**Pull request by strimo378**
1. The transformation learned to merge templated base classes
2. The transformation was splitted into two transformation: (1) remove-base-class only removes the base class, (2) merge-base-class removes the base class and copies the content of the base class

### CODE REVIEW

**Comment by marxin** [comment]
Please fix the issue with older LLVM, and please add a comment to `getDescribedTemplateParams` mentioning which LLVM version needs the function.

**Comment by strimo378** [comment]
I have no idea when getDescribedTemplateParams was introduced. Do you have an idea how to find out?

**Comment by strimo378** [comment]
Can we deactivate the test case for clang9?

**Comment by marxin** [comment]
> Do you have an idea how to find out?

```
git branch --all --contains 2a3b86c157166f3b15f718443334ab0e27b40592 | grep release
  remotes/origin/release/11.x
  remotes/origin/release/12
  remotes/origin/release/12.x
  remotes/origin/release/13.x
  remotes/origin/release/14.x
  remotes/origin/release/15.x
```

So it's there likely since `11.x`. So please document that it's not needed since `LLVM 11`. For now, I would like to keep support for LLVM 9 as openSUSE Leap uses it.

**Comment by marxin** [comment]
> Can we deactivate the test case for clang9?

Yes, please take a look at e5c9c380d1fc05bb2e11003b7d135897ce824da8.

**Comment by marxin** [comment]
Please rebase on top of 48a8e24bfda0192084505e0ba2cebfab3e65f7a6, I fixed there LLVM version detection.

File: clang_delta/tests/test_clang_delta.py [hunk 1]
```diff
@@ -618,6 +618,7 @@ def test_rename_operator_test2(self):
     def test_merge_base_class_test1(self):
         self.check_clang_delta('merge-base-class/test1.cc', '--transformation=merge-base-class --counter=1')
 
+    @unittest.skipIf(get_llvm_version() <= 9, 'Fails with LLVM 8')
```
Comment (marxin) [inline clang_delta/tests/test_clang_delta.py:621]: I would use 'Fails with LLVM <= 9', please adjust it.

**Review by marxin — CHANGES_REQUESTED** [review]

**Comment by strimo378** [comment]
So ... rebased ... are 3 commits ok, or should I combine them?

**Comment by marxin** [comment]
Please squash them into one.

**Comment by marxin** [comment]
Next time, please use "Convert to draft" instead of renaming ;)

**Comment by strimo378** [comment]
Thx for the hint. It took me 5 min to find it on the page :)

**Comment by marxin** [comment]
> It took me 5 min to find it on the page :)

Yeah, it's a pretty small button.

**Pull request merged**